unstructured_ingest/v2/pipeline/interfaces.py

```diff
@@ -132,7 +134,7 @@ def _run(self, fn: Callable, **kwargs: Any) -> Optional[Any]:
     async def _run_async(self, fn: Callable, **kwargs: Any) -> Optional[Any]:
         raise NotImplementedError
 
-    def run(self, _fn: Optional[Callable] = None, **kwargs: Any) -> Optional[Any]:
+    def run(self, _fn: Callable[..., Any] | None = None, **kwargs: Any) -> Optional[Any]:
```

Comment:
We can't support the `type | None` syntax because we need to support python3.9 when that didn't exist.

Reply:
I believe the `from __future__ import annotations` line at the top of the file resolves that. I've been using that syntax in `unstructured` where we also support 3.9. Will look it up in the docs, though.

Reply:
This [article](https://blog.derlin.ch/python-type-hints-and-future-annotations) has a decent explanation. As of py 3.7 and [PEP 563](https://peps.python.org/pep-0563/), the evaluation of annotations is deferred and treated as strings. It seems the only thing we have to be careful of is that "future.annotations lets you use anything as annotations, including features that are not yet available (or even "garbage"). This is very dangerous if you plan on reading types at runtime." But I assume our tests in 3.9 would cover that.

Reply:
Great! I wasn't aware of that.